app/assets/javascripts/ndt_speed_test.coffee

```diff
@@ -1,42 +1,60 @@
 bind_ndt_speed_calculation = ->
-  ndtServer = undefined
-  ndtPort = '3010'
-  ndtProtocol = 'wss'
-  ndtPath = '/ndt_protocol'
-  ndtUpdateInterval = 1000
-
-  getNdtServer = ->
-    xhr = new XMLHttpRequest
-    mlabNsUrl = 'https://mlab-ns.appspot.com/ndt_ssl?format=json'
-    xhr.open 'GET', mlabNsUrl, true
-    xhr.send()
-
-    xhr.onreadystatechange = ->
-      if xhr.readyState == 4
-        if xhr.status == 200
-          ndtServer = JSON.parse(xhr.responseText).fqdn
-          ndtServerIp = JSON.parse(xhr.responseText).ip
-
-          console.log 'Using M-Lab Server ' + ndtServer
-          document.getElementById('submission_hostname').value = ndtServer;
-
-          # Track NDT server used
-          Sentry.setExtra("mlab_server", ndtServer)
-        else
-          console.log 'M-Lab NS lookup failed.'
-
-          # Track NDT server used
-          Sentry.setExtra("mlab_server", 'failed')
-      return
-    return
-
-  getNdtServer()
 
   NDT_meter = new NDTmeter('#ndt-svg')
   $('#start_ndt_test').on 'click', ->
-    NDT_client = new NDTjs(ndtServer, ndtPort, ndtProtocol, ndtPath, NDT_meter, ndtUpdateInterval)
-    NDT_client.startTest()
-    return
+    # Note: rates are in kbits/s, latency is in milliseconds.
+    s2cRate = undefined
+    c2sRate = undefined
+    minRTT = undefined
+    NDT_meter.onstart()
+    ndt7.test {
```

Comment:
It's a subjective stylistic thing, but would you add parentheses for this function (`ndt7.test({...}, {...})`). Might be easier to read since it has multiple arguments over multiple lines.

Reply:
No worries! CoffeScript isn't my favorite either. We're planning to eventually move to TypeScript, but this stack is a holdover from the code we forked.

For me, it worked to add an open paren on line 10 between the method name and the open brace:
```coffescript
    ndt7.test({
```

And then a close paren after the closing brace on line 57:
```coffescript
    })
```

We also don't have linting set up for this repo currently. If you're seeing that error in your code editor, is it possible that linter was bundled with/run by your editor? Does the site build and work for you in the browser if you make that change?

I'm just making a guess here, but it's also possible that the linter would be happier with different placement of line breaks. Maybe it doesn't like closing the first object and opening the second object on the same line, so would it like this kind of thing better?

```coffeescript
ndt7.test(
  {
    // ...
  },
  {
    // ...
  }
)
```

Reply:
Strange - I definitely had CoffeeLint enabled in my editor, but I also get the same error when I open http://localhost:3000 after restarting the frontend. With parens on lines 10 and 57:

```
ExecJS::RuntimeError in Home#index

Showing /suyc/app/views/home/_introduction.html.erb where line #6 raised:

SyntaxError: [stdin]:14:6: unexpected ,
```

I've also tried different placement of line breaks, but apparently a newline after `ndt7.test(` is a syntax error in itself.
In the browser:
```
SyntaxError: [stdin]:10:16: unexpected newline
```

Reply:
Strike that, it was the extra space between `ndt7.test` and the opening parenthesis, and the error message didn't help much. I've committed the change. :)